x-pack/filebeat/input/awss3/input.go

```diff
@@ -48,15 +50,32 @@ func (im *s3InputManager) Create(cfg *conf.C) (v2.Input, error) {
 		return nil, fmt.Errorf("initializing AWS config: %w", err)
 	}
 
+	// The awsConfig now contains the region from the credential profile or default region
+	// if the region is explicitly set in the config, then it wins
+	if config.RegionName != "" {
+		awsConfig.Region = config.RegionName
+	}
+
 	if config.AWSConfig.Endpoint != "" {
-		// Add a custom endpointResolver to the awsConfig so that all the requests are routed to this endpoint
-		awsConfig.EndpointResolverWithOptions = awssdk.EndpointResolverWithOptionsFunc(func(service, region string, options ...interface{}) (awssdk.Endpoint, error) {
-			return awssdk.Endpoint{
-				PartitionID:   "aws",
-				URL:           config.AWSConfig.Endpoint,
-				SigningRegion: awsConfig.Region,
-			}, nil
-		})
+		// Parse a URL for the host regardless of it missing the scheme
+		endpointUri, err := url.Parse(config.AWSConfig.Endpoint)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse endpoint: %w", err)
+		}
+
+		// For backwards compat:
+		// If the endpoint does not start with S3, we will use the endpoint resolver to make all SDK requests use the specified endpoint
+		// If the endpoint does start with S3, we will use the default resolver uses the endpoint field but can replace s3 with the desired service name like sqs
+		if !strings.HasPrefix(endpointUri.Hostname(), "s3") {
```

Comment:
I wonder if this should check to see if s3 is in the URL at all -- as we probably want the same behavior for vpce endpoints like `vpce.s3....`

Reply:
If that check was removed then it seems like the only fundamental change in this function is adding `Source:        awssdk.EndpointSourceCustom,` to the existing resolver. The big question for me is still whether this whole check can be removed in lieu of the existing endpoint handling in the config modifiers (or vice versa).